### PR TITLE
HCAL DQM histogram booking update

### PR DESCRIPTION
**Pull request by spandeyehep**
This PR mainly concerns the HCAL DQM histogram booking script. 
More details can be found out in the discussion of this[1] RelVal HN thread.

Involves following packages:
DQMOffline/Hcal
Validation/CaloTowers
Validation/HcalHits

[1] https://hypernews.cern.ch/HyperNews/CMS/get/relval/10906.html

### CODE REVIEW

**Comment by andrius-k** [comment]
Could you please check if the large InputRelVal.json file is required. Also could you please check that html files are not autogenerated and should actually be checked in.

**Comment by spandeyehep** [comment]
Hi @andrius-k , 
InputRelVal.json file is required for the HCAL offline DQM relval plots (it defines ranges etc. for the plots).
And html files are not auto-generated, these are manually written skeletons.

**Comment by hatakeyamak** [comment]
Shubham already wrote basically, but we should point out that largish changes under
Validation/CaloTowers/test/macros really don't interfere with the central DQM/validation operation. Something we use for plotting offline from centrally harvested DQM files. So, they don't have to be in the central cmssw repository in a sense, but we prefer to have CMSSW Validation/CaloTowers/test/macros location updated with our latest macros for book-keeping.

Other changes are simply:
+ book1D -> book1DD for some histograms where double seems necessary
+ reverting the hep17 specific option (introduced for special operation in 2017) to false for 2018 and future scenarios

**Comment by andrius-k** [comment]
Hi,

I ran the instructions and everything went well except for one thing. When executing python scripts directly (`./script.py` instead of `python ./script.py`) they would pick up the wrong python because it is declared in the first line of the file. I think first line of every file should be changed to `#!/usr/bin/env python` so that correct python is used when running them.
The same error happened when running `.csh` files that run python files internally.

**Comment by spandeyehep** [comment]
Hi @andrius-k , Thanks for the suggestion. I have updated the PR.

**Comment by andrius-k** [comment]
+1
Fine for DQM. @fabiocos you might want to check if this PR is fine with CMSSW policies.

**Comment by schneiml** [comment]
@fabiocos we (Andrius and me) tried the instructions provided with the PR (that is why we asked for them) and things seem to work as expected, so I'd say that counts as a standalone test.

What this PR adds is for sure better than the outdated scripts that are currently in this folder.

File: Validation/CaloTowers/test/macros/RelValHarvest.py [hunk 1]
```diff
@@ -1,104 +1,212 @@
...
+                 ofnBlank = "HcalRecHitValidationRelVal_%(sample)s_%(label)s_%(info)s.root",
+                 label = "CMSSW_X_Y_Z",
+                 slabel = "XYZ",
+                 X509_CERT_DIR = os.getenv('X509_CERT_DIR', "/etc/grid-security/certificates"),
+                 X509_USER_PROXY = os.getenv('X509_USER_PROXY'),
+                 relValDIR = "CMSSW_?_?_x",
+                 printDS = False,
+                 camType = "MC"):
+               
+    print "Taking filenames from directory %s"%relValDIR
```
[the remark this answers is not in the document]
Comment (spandeyehep) [reply Validation/CaloTowers/test/macros/RelValHarvest.py]: Hi @fabiocos , I have updated the python scripts to make it python3 compatible.

**Comment by andrius-k** [comment]
+1

File: DQMOffline/Hcal/python/HLTHcalRecHitParam_cfi.py [hunk 2]
```diff
@@ -17,3 +17,9 @@
 run2_HEPlan1_2017.toModify(hltHCALRecHitsAnalyzer,
       hep17 = cms.untracked.bool(True)
 )
+
+from Configuration.Eras.Modifier_run2_HE_2018_cff import run2_HE_2018
```
[the remark this answers is not in the document]
Comment (abdoulline) [reply DQMOffline/Hcal/python/HLTHcalRecHitParam_cfi.py]: Kevin, do you mean that for 2018  "hep17" gets back to False (primary config setting) just "for free"?

[the remark this answers is not in the document]
Comment (kpedro88) [reply DQMOffline/Hcal/python/HLTHcalRecHitParam_cfi.py]: Yes, in the Run2_2018 Era, the run2_HEPlan1_2017 modifications are never activated.

**Comment by abdoulline** [comment]
@kpedro88 
Kevin, do you think this redundancy of"hep17" for  run2_HE_2018 
has to be removed (and then once again tested and signed) to get it merged or it can go as is?

**Comment by kpedro88** [comment]
It should be removed.

**Comment by abdoulline** [comment]
OK, I believe Shubham has all in hands to do that.

**Comment by spandeyehep** [comment]
Hi @kpedro88 @abdoulline ,
I have updated the PR.
Thanks.

**Comment by kpedro88** [comment]
please test

**Comment by andrius-k** [comment]
+1

**Comment by kpedro88** [comment]
+1

**Pull request merged**